### ISSUE TITLE
find titles in embeded epubs

### ISSUE THREAD

**Issue by anarcat** [koreader/koreader#2620]
It would be nice if Koreader could read the content of ePUB files and find the title. I have files that are automatically generated here and the filename is a meaningless identifier. The default Kobo firmware correctly shows the book title in the file listing, couldn't Koreader do the same?

**Comment by KenMaltby** [comment]
How do they appear in calibre?

**Comment by anarcat** [comment]
The title from the ePUB shows up correctly... not sure what you are asking here.

**Comment by KenMaltby** [comment]
If caliber is showing the titles as you want (wherever it is extracted from, Metadata?) you should be able to have calibre generate a file with that name, that KOReader will use.  EPUB to EPUB conversion in calibre can be a cure for many problem files.

**Comment by anarcat** [comment]
yeah, i'm looking into other solutions as well - *i* am the one generating those files after all... But i was hoping that koreader could be a little smarter: there's more than the title in there... Author and other metadata could be shown on multiple lines, for example, something that would be quite hard to do in a single filename (which *can* contain newlines, but is suspect that could break all sorts of other weird stuff).

anyways, I understand if koreader would not parse the epub for performance reasons, for example. i just wanted to see if that was something that could be considered at all...

in other words, feel free to close this issue if you think this feature shouldn't be implemented...

**Comment by Hzj-jie** [comment]
In the issue you referred here, I believe the most important issue is the cost of reading metadata from files.

**Comment by houqp** [comment]
We can experiment this feature with the new koreader home to see how much performance hit we will get. Ideally, if we cache the file name in the document config, the overhead should be really low.

**Comment by Frenzie** [comment]
Closing in favor of 2614.

**Comment by anarcat** [comment]
i guess you meant to link to #2614 here.
